### PR TITLE
add tx origin blacklist to RFQ options

### PR DESCRIPTION
**Pull request by alexkroeger**
## Description

Add the ability to blacklist specified `txOrigin`s from using RFQT.

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/asset-swapper/src/swap_quoter.ts [hunk 1]
```diff
@@ -410,6 +410,14 @@ export class SwapQuoter {
         return whitelistedApiKeys.includes(apiKey);
     }
 
+    private _isTxOriginBlacklisted(txOrigin: string | undefined): boolean {
+        if (!txOrigin) {
+            return false;
+        }
+        const blacklistedApiKeys = this._rfqtOptions ? this._rfqtOptions.txOriginBlacklist : [];
```
Comment (alexkroeger) [inline packages/asset-swapper/src/swap_quoter.ts:417]: should this be a set?

Comment (PirosB3) [reply packages/asset-swapper/src/swap_quoter.ts:417]: Yes, or a `Set` or a HashSet (or object, in JS)

File: packages/asset-swapper/src/constants.ts [hunk 2]
```diff
@@ -52,6 +52,7 @@ const DEFAULT_SWAP_QUOTER_OPTS: SwapQuoterOpts = {
     rfqt: {
         takerApiKeyWhitelist: [],
         makerAssetOfferings: {},
+        txOriginBlacklist: [],
```
Comment (PirosB3) [inline packages/asset-swapper/src/constants.ts:55]: Could you make this a set?

File: packages/asset-swapper/src/swap_quoter.ts [hunk 3]
```diff
@@ -410,6 +410,14 @@ export class SwapQuoter {
         return whitelistedApiKeys.includes(apiKey);
     }
 
+    private _isTxOriginBlacklisted(txOrigin: string | undefined): boolean {
+        if (!txOrigin) {
```
Comment (PirosB3) [inline packages/asset-swapper/src/swap_quoter.ts:414]: Are you sure that txOrigin can be undefined?

**Review by PirosB3 — APPROVED** [review]
Missing tests, otherwise it's awesome!

File: packages/asset-swapper/src/swap_quoter.ts (hunk 3 above)
Comment (alexkroeger) [reply packages/asset-swapper/src/swap_quoter.ts:414]: it's explicitly checked for later in the code block

File: packages/asset-swapper/src/swap_quoter.ts [hunk 4]
```diff
@@ -410,6 +410,14 @@ export class SwapQuoter {
...
+            return false;
+        }
+        const blacklistedTxOrigins = this._rfqtOptions ? this._rfqtOptions.txOriginBlacklist : new Set();
```
Comment (PirosB3) [inline packages/asset-swapper/src/swap_quoter.ts:417]: Without having to create a `new Set` every time this fn is called, can we change something like
```
if (this._rfqtOptions && this._rfqtOptions.txOriginBlacklist && this._rfqtOptions.txOriginBlacklist.has(txOrigin.toLowerCase())) {
     return true;
}
return false;
```

this is optional, if it makes sense to you

File: packages/asset-swapper/src/swap_quoter.ts [hunk 5]
```diff
@@ -410,6 +410,14 @@ export class SwapQuoter {
...
+        return blacklistedTxOrigins.has(txOrigin.toLowerCase());
```
Comment (PirosB3) [inline packages/asset-swapper/src/swap_quoter.ts:418]: good catch, I had forgotten about case sensitivity

**Review by PirosB3 — APPROVED** [review]

File: packages/asset-swapper/src/swap_quoter.ts (hunk 5 above)
Comment (alexkroeger) [reply packages/asset-swapper/src/swap_quoter.ts:418]: 👍

**Pull request merged**